supreme_court_predictions/statistics/descriptives.py

```diff
@@ -0,0 +1,324 @@
+        avg_speakers = len(
+            utterances.loc[:, ["case_id", "speaker"]].value_counts()
+        ) / len(utterances.loc[:, "case_id"].unique())
+
+        descriptives = {
+            "num of utterances": avg_utterance,
+            "num of speakers": avg_speakers,
+        }
+
+        utterances_descriptives = pd.DataFrame.from_dict(
+            descriptives, orient="index", columns=["average"]
+        )
+
+        return utterances_descriptives
+
+    def get_voters_desc(self):
+        """
+        Calculates descriptive statistics for the voters DataFrame.
+
+        :return DataFrame of descriptive statistics.
+        """
+        # Load data
+        voters = pd.read_csv(self.local_path + "voters_df.csv")
+
+        # Get descriptive stats of votes
+        voter_stats = self.get_count_desc(voters, ["vote"])
+        voter_stats.loc[("justices", ""), :] = len(
+            voters.loc[:, "voter"].unique()
+        )
+
+        return voter_stats
+
+    def parse_all_data(self):
+        """
+        Calculates descriptive statistics for all of the supreme corpus
+        DataFrames and exports them to csv/excel (if applicable).
+        """
+
+        print("Grabbing advocates descriptive statistics...")
+        self.advocates_stats = self.get_advocate_desc()
+
+        print("Grabbing conversations descriptive statistics...")
+        self.conversations_stats = self.get_conversation_desc()
+
+        print("Grabbing speakers descriptive statistics...")
+        self.speakers_stats = self.get_speaker_desc()
+
+        print("Grabbing voters descriptive statistics...")
+        self.voters_stats = self.get_voters_desc()
+
+        print("Grabbing utterances descriptive statistics...")
+        self.utterances_stats = self.get_utterances_desc()
+
+        if self.save_data:
+            # Outputting to CSVs
+            descriptives = [
+                self.advocates_stats,
+                self.conversations_stats,
+                self.speakers_stats,
+                self.voters_stats,
+                self.utterances_stats,
+            ]
+            outpaths = [
+                self.output_path + "/advocates_stats.csv",
+                self.output_path + "/conversations_stats.csv",
+                self.output_path + "/speakers_stats.csv",
+                self.output_path + "/voters_stats.csv",
+                self.output_path + "/utterances_stats.csv",
+            ]
+
+            for idx, desc in enumerate(descriptives):
+                desc.to_csv(outpaths[idx], index=True, encoding=ENCODING_UTF_8)
+
+            # Outputting to a single excel
+            desc_out = self.output_path + "/descriptive_statistics.xlsx"
+            # Known ExcelWriter issue:
+            # https://github.com/pylint-dev/pylint/issues/3060
+            # pylint: disable=abstract-class-instantiated
+            with pd.ExcelWriter(desc_out) as writer:
+                # pylint: enable=abstract-class-instantiated
```

Comment:
I added these comments to escape a linting issue: https://github.com/pylint-dev/pylint/issues/3060

cc: @chanteriam